### DISCUSSION TITLE
Ability to alias filter column names

### DISCUSSION THREAD

**Discussion by ant-magic42** [tailflow/laravel-orion#295] [Ideas]
Laravel API resources provide the ability to alias database fields in the JSON response, e.g.
```
public function toArray($request):
{
    return [
        'currency_id' => $this->currency_id,
        'currency_name' => $this->curr_name, // currency_name is nicer than curr_name
    ];
}
```

And Laravel Orion provides [Filters](https://tailflow.github.io/laravel-orion-docs/v2.x/guide/search.html#filtering) which expose the raw database fields, e.g.

```
public function filterableBy() : array
{
    return ['curr_name'];
}
```

But this means that the frontend developer has a "nice" name in the JSON response but must use a "raw" database field name in the filter.

So it would be helpful to the frontend developer if it was possible to expose `filterableBy` attributes via an alias, e.g.

```
public function filterableBy() : array
{
    return [
        'currency_name' => 'curr_name',  // perhaps this could be a way to specify a filter alias?
    ];
}
```

Does this suggestion make sense?  Or is there a better way?

**Comment by alexzarbn** [comment]
Hi @anthonymf,

Why not just rename the column in db? 🤔 Column names should be easy to understand on both frontend and backend.

**Comment by ant-magic42** [comment]
@alexzarbn Renaming the db column would not suit use cases where the DB is tightly coupled to other systems or other existing code in the Laravel application.

But I think the principle of Laravel API Resources is that they allow fields in JSON responses to be aliased in the response, rather than having to do this in the database. This neatly decouples the API responses from the db structure.
I’m suggesting that an ability to alias fields in ‘filterableBy‘ would be useful in the same way.

**Comment by alexzarbn** [comment]
@anthonymf Okay, that makes sense. I think introducing a method `aliases` that lists such fields would work. It can be used by `QueryBuilder` for filtering and sorting, and by controllers to map incoming request attributes onto model attributes.

**Comment by ant-magic42** [comment]
@alexzarbn Yes, that sounds like a good plan.
